src/MOI_wrapper.jl

```diff
@@ -345,33 +346,27 @@ function MOI.supports(model::Optimizer, ::MOI.VariablePrimalStart,
     return MOI.supports(model.optimizer, MOI.VariablePrimalStart(), MOI.VariableIndex)
 end
 
-function MOI.get(model::Optimizer, attr::MOI.VariablePrimal, vi::MOI.VariableIndex)
+function MOI.get(model::Optimizer, attr::MOI.VariablePrimal, vi::VI)
     MOI.check_result_index_bounds(model.optimizer, attr)
     return MOI.get(model.optimizer, attr, vi)
 end
 
-function MOI.add_constraint(model::Optimizer, f::MOI.SingleVariable, s::SUPPORTED_SCALAR_SETS)
-    ci = MOI.add_constraint(model.optimizer, f, s)
-    push!(model.con_idx, ci)
-    return ci
-end
-
 function MOI.delete(model::Optimizer, ci::CI{F,S}) where {F <: SUPPORTED_SCALAR_FUNCTIONS, S <: SUPPORTED_SCALAR_SETS}
     filter!(e -> e≠ci, model.con_idx)
     MOI.delete(model.optimizer, ci) 
 end
 
-function MOI.delete(model::Optimizer, ci::CI{F,S}) where {F <: MOI.VectorAffineFunction{Float64}, S <: SUPPORTED_VECTOR_SETS}
+function MOI.delete(model::Optimizer, ci::CI{F,S}) where {F <: SUPPORTED_VECTOR_FUNCTIONS, S <: SUPPORTED_VECTOR_SETS}
     filter!(e -> e≠ci, model.con_idx)
```

Comment:
```suggestion
    filter!(≠(ci), model.con_idx)
```

`≠(ci)` creates an anonymous function-like object that does the same thing as x -> x != y

Reply:
thanks! i'll read up more on it

Reply:
hey. i wasn't able to execute this. even a simple script like
```julia
a = [1,2,3]
x = 2
filter(≠(x), a)
```
with 
```
a = [1,2,3]
x = 2
filter!(≠(x), a)
a = [1,2,3]
x = 2
filter!(≠(x), a)
MethodError: no method matching !=(::Int64)
Closest candidates are:
  !=(::Any, !Matched::Any) at operators.jl:185
  !=(!Matched::Float64, !Matched::Float64) at float.jl:450
  !=(!Matched::Float32, !Matched::Float32) at float.jl:449
  ...
```